docs/manual/scala/guide/services/code/AdditionalRouters.scala

```diff
@@ -0,0 +1,88 @@
+package docs.scaladsl.services
+
+package helloservice {
+
+  import com.lightbend.lagom.scaladsl.api.ServiceCall
+  import scala.concurrent.Future
+
+
+  //#hello-service
+  trait HelloService extends Service {
+
+    def hello(id: String): ServiceCall[NotUsed, String]
+
+    override final def descriptor = {
+      import Service._
+      named("hello")
+        .withCalls(
+          pathCall("/api/hello/:id", hello _).withAutoAcl(true)
+        )
+        .withAcls(
+          // extra ACL to expose additional router endpoint on ServiceGateway
+          ServiceAcl(pathRegex = Some("/api/files"))
+        )
+    }
+  }
+  //#hello-service
+
+  //#hello-service-impl
+  class HelloServiceImpl extends HelloService {
+    override def sayHhello(name: String) = ServiceCall { _ =>
+      Future.successful(s"Hello $name!")
+    }
+  }
+  //#hello-service-impl
+}
+
+package lagomapplication {
+
+  import helloservice._
+
+  import com.lightbend.lagom.scaladsl.server._
+  import play.api.libs.ws.ahc.AhcWSComponents
+  import com.softwaremill.macwire._
+
+
+  class SomePlayRouter extends SimpleRouter{
+    override def routes: Routes = ???
+  }
+
+  abstract class HelloApplication(context: LagomApplicationContext)
+    extends LagomApplication(context)
+      with AhcWSComponents {
+
+    //#lagom-application-some-play-router
+    override lazy val lagomServer =
+        serverFor[HelloService](wire[HelloServiceImpl])
+        .additionalRouter(wire[SomePlayRouter])
+    //#lagom-application-some-play-router
+  }
+
+  //#file-upload-router
+  import play.api.mvc.{DefaultActionBuilder, Results}
+  import play.api.routing.Router
+  import play.api.routing.sird._
+
+  class FileUploadRouter(action: DefaultActionBuilder) {
+    val router = Router.from {
+      case POST(p"/api/files") => action { _ =>
+        // for the sake of simplicity, this implementation
```

Comment:
Same as the Java version. At least access the file to ensure the examples does what it says it does. You can do:

```scala
case POST(p"/api/files") => action { req =>
  val body = req.body.asMultipartFormData
  ...
}
```

Or inject `PlayBodyParsers` and do:

```scala
case POST(p"/api/files") => action(parsers.multipartFormData) { req =>
  val multipartBody = req.body
  ...
}
```

Reply:
Initially, I had an example that indeed accessed the file,  but I reverted after discussing with @ignasi35. We kind of decided to make it minimal.

I will fix the other comments and then implement this one so we can compare. 

The Scala variant has some more boiler-plate because `req.body.asMultipartFormData` returns an `Option`.